crates/core/src/network/state.rs

```diff
@@ -85,4 +108,96 @@ impl NetworkState {
             HolochainError::ErrorGeneric("Network not initialized".to_string()),
         )
     }
+
+    pub(crate) fn cache_validation(&mut self, agent: Address, validation: &ValidationPackage) {
+        if validation.source_chain_headers.is_none() {
+            return;
+        }
+        if self.validation_cache.contains_key(&agent) {
+            let mut cache_entry = self.validation_cache.get_mut(&agent).unwrap();
+            if validation.chain_header.timestamp() > cache_entry.latest_header().timestamp() {
+                debug!("Vcache: updating {}:{:?}", &agent, validation.chain_header);
+                debug!(
+                    "header count: {}",
+                    validation.source_chain_headers.as_ref().unwrap().len()
+                );
+                cache_entry.headers = validation.source_chain_headers.clone().unwrap();
+                if validation.source_chain_entries.is_some() {
+                    debug!(
+                        "entry count: {}",
+                        validation.source_chain_entries.as_ref().unwrap().len()
+                    );
+                    cache_entry.entries = validation.source_chain_entries.clone();
+                }
+            } else {
+                debug!("Vcache: already {}:{:?}", &agent, validation.chain_header);
+            }
+        } else {
+            debug!(
+                "Vcache: initial insert {}:{:?} {:#?}",
+                &agent, validation.chain_header, &validation
+            );
+            self.validation_cache.insert(
+                agent,
+                ValidationCacheData {
+                    entries: validation.source_chain_entries.clone(),
+                    headers: validation.source_chain_headers.as_ref().unwrap().clone(),
+                    cached_at: SystemTime::now(),
```

Comment:
Are we cool keeping these things cached forever? `cached_at` doesn't seem to ever be read.

Reply:
I'm not persisting this data, so it will be cleared on reboot, which is fine I think because in that case the chain in the cache is probably pretty stale and will need to be fetched anyways.  I think he only real reason to dump the cache is because memory gets tight...